salt/utils/event.py

```diff
@@ -514,7 +516,10 @@ def destroy(self, linger=5000):
             # Wait at most 2.5 secs to send any remaining messages in the
             # socket or the context.term() below will hang indefinitely.
             # See https://github.com/zeromq/pyzmq/issues/102
-            self.sub.setsockopt(zmq.LINGER, linger)
+            try:
```

Comment:
Actually, how do you get that int error? If you look at the __del__ function it *already* has the try/except block around it.